File: ideal-core/src/main/java/cn/idealframework2/event/EventListenerRegistrar.java
package cn.idealframework2.event;

import javax.annotation.Nonnull;

/**
 * @author 宋志宗 on 2022/11/8
 */
public interface EventListenerRegistrar {

  /**
   * 注册事件监听器
   *
   * @param registry 事件监听注册中心
   * @author 宋志宗 on 2022/11/8
   */
  void registerEventListener(@Nonnull EventListenerRegistry registry);
}

File: ideal-core/src/main/kotlin/cn/idealframework2/event/coroutine/EventListenerRegistrar.kt
package cn.idealframework2.event.coroutine

/**
 * @author 宋志宗 on 2022/11/8
 */
interface EventListenerRegistrar {

  /**
   * 注册事件监听器
   *
   * @param registry 事件监听注册中心
   * @author 宋志宗 on 2022/11/8
   */
  fun registerEventListener(registry: EventListenerRegistry)
}
